### DISCUSSION TITLE
Allow using `LIKE` keyword with `Custom Keys & Scoping`

### DISCUSSION THREAD

**Discussion by mahmoudmohamedramadan** [laravel/framework#49133] [Ideas]
Welcome Folks! 👋

When I was making a review of [Custom Keys & Scoping](https://laravel.com/docs/10.x/routing#implicit-model-binding-scoping) section. I wrote the next code to ensure that everything was functioning properly

```PHP
// routes\web.php

Route::get(
    'without-scope-binding/users/{user}/posts/{post}',
    [UserPostController::class, 'withoutScopeBinding']
);

Route::get(
    'implicit-scope-binding/users/{user}/posts/{post:title}',
    [UserPostController::class, 'ImplicitScopeBinding']
);
```

```PHP
// App\Http\Controllers\Dashboard

class UserPostController extends Controller
{
    public function withoutScopeBinding(User $user, Post $post)
    {
        dd("User Name Is: {$user->name}, and the Post Title Is: {$post->title}");
    }

    public function ImplicitScopeBinding(User $user, Post $post)
    {
        dd("User Name Is: {$user->name}, and the Post Title Is: {$post->title}");
    }
}
```

When I tested the first route and wrote `/without-scope-binding/users/1/posts/1`, I found that Laravel returns the name of the first user's id and the title of the first post without assuming that the first user has the `posts` method, and when I tested the second route and pass `/implicit-scope-binding/users/1/posts/Excepturi nemo.` Laravel gets a valid value back also, here I told to myself surely I'll not remember the exact and full name of the post's title so, I suggest here to search using `LIKE` keyword, I think that it will be a very nice to add this feature to Laravel.

> [!IMPORTANT]
> I think that it will not be a BC because I will allow the `resolveRouteBindingQuery` method to accept the operator value with a default `=` value, and I'll use `LIKE` in case the type of the given field is a string.

**What do you think about that idea, I'm waiting for your opinion 🧐.**

**Comment by shaedrich** [comment]
Since it is already possible to use `LIKE` directly in the `resolveRouteBindingQuery` method, it's more a question of whether we want to give this functionality to the Route class itself, did I get that correctly? So in this case, the only complication (mentioned in your BC note) would be that the `resolveRouteBindingQuery` method isn't directly called by the user so this information has to passed through somehow which might require changes in several places, which, of course, isn't a problem per se.

**Reply by mahmoudmohamedramadan** [reply]
Yup, that is what I said, but I don't know what the opinion of @taylorotwell about that feature is.
